tests/unit/moduleapi/moduleconfigs.tcl

```diff
@@ -0,0 +1,186 @@
+        assert_equal [r config get maxclients] "maxclients 10000"
+        # test we can't set other module's configs
+        r module load $testmoduletwo
+        catch {[r module loadex $testmodule CONFIG configs.test no]}
+        assert_match {*ERR*} $e
+        assert_equal [r config get configs.test] "configs.test yes"
+        r module unload configs
+    }
+
+    test {test config rewrite with dynamic load} {
+        #translates to: super \0secret password
+        r module loadex $testmodule CONFIG moduleconfigs.string \x73\x75\x70\x65\x72\x20\x00\x73\x65\x63\x72\x65\x74\x20\x70\x61\x73\x73\x77\x6f\x72\x64 ARGS
+        assert_equal [lindex [lindex [r module list] 0] 1] moduleconfigs
+        assert_equal [r config get moduleconfigs.string] "moduleconfigs.string {super \0secret password}"
+        r config set moduleconfigs.mutable_bool yes
+        r config set moduleconfigs.memory_numeric 750
+        r config set moduleconfigs.enum two
+        r config rewrite
+        restart_server 0 true false
+        # Ensure configs we rewrote are present
+        assert_equal [r config get moduleconfigs.mutable_bool] "moduleconfigs.mutable_bool yes"
+        assert_equal [r config get moduleconfigs.memory_numeric] "moduleconfigs.memory_numeric 750"
+        assert_equal [r config get moduleconfigs.string] "moduleconfigs.string {super \0secret password}"
+        assert_equal [r config get moduleconfigs.enum] "moduleconfigs.enum two"
+        assert_equal [r config get moduleconfigs.numeric] "moduleconfigs.numeric -1"
+        r module unload moduleconfigs
+    }
+
+    test {test multiple modules with configs} {
+        r module load $testmodule
+        r module loadex $testmoduletwo CONFIG configs.test yes
+        assert_equal [r config get moduleconfigs.mutable_bool] "moduleconfigs.mutable_bool yes"
+        assert_equal [r config get moduleconfigs.immutable_bool] "moduleconfigs.immutable_bool no"
+        assert_equal [r config get moduleconfigs.memory_numeric] "moduleconfigs.memory_numeric 1024"
+        assert_equal [r config get moduleconfigs.string] "moduleconfigs.string {secret password}"
+        assert_equal [r config get moduleconfigs.enum] "moduleconfigs.enum one"
+        assert_equal [r config get moduleconfigs.numeric] "moduleconfigs.numeric -1"
+        assert_equal [r config get configs.test] "configs.test yes"
+        r config set moduleconfigs.mutable_bool no
+        r config set moduleconfigs.string nice
+        r config set moduleconfigs.enum two
+        r config set configs.test no
+        assert_equal [r config get moduleconfigs.mutable_bool] "moduleconfigs.mutable_bool no"
+        assert_equal [r config get moduleconfigs.string] "moduleconfigs.string nice"
+        assert_equal [r config get moduleconfigs.enum] "moduleconfigs.enum two"
+        assert_equal [r config get configs.test] "configs.test no"
+        r config rewrite
+        restart_server 0 true false
```

Comment:
so this part tests loading modules and configs, from the config file, right?
mabe add a comment, since whoever misses it may think this is completely uncovered.

Reply:
actually, this way we have no way to test what happens when loading a module with configuration error from config file.
let's add a test that loads a module from config file.
hints:
1. `testrdb.tcl` does it, you can copy some code from there.
2. the test infra doesn't support passing the same config to `start_server` twice, so don't try to test two modules from a config file.

please also add tests for rejected configs, and missing LoadConfig calls for this path.